test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,329 @@
+
+	cfg := server.BaseConfig(t)
+
+	var err error
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+
+	export := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-export",
+		},
+	}
+
+	slice := &apisv1alpha1.APIExportEndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "my-slice",
+		},
+		Spec: apisv1alpha1.APIExportEndpointSliceSpec{
+			APIExport: apisv1alpha1.ExportBindingReference{
+				Path: exportClusterPath.String(),
+				Name: export.Name,
+			},
+		},
+	}
+
+	partition := &topologyv1alpha1.Partition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-partition",
+		},
+		Spec: topologyv1alpha1.PartitionSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"region": "apiexportendpointslice-test-region",
+				},
+			},
+		},
+	}
+
+	t.Logf("Creating an APIExportEndpointSlice with reference to a nonexistent APIExport")
+	sliceClient := kcpClusterClient.ApisV1alpha1().APIExportEndpointSlices()
+
+	_, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+	require.True(t, apierrors.IsForbidden(err), "no error creating APIExportEndpointSlice (admission should have declined it)")
+
+	t.Logf("Creating the missing APIExport")
+	exportClient := kcpClusterClient.ApisV1alpha1().APIExports()
+	_, err = exportClient.Cluster(exportClusterPath).Create(ctx, export, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating APIExport")
+
+	t.Logf("Retrying to create the APIExportEndpointSlice after the APIExport has been created")
+	require.Eventually(t, func() bool {
+		slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		return err == nil
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected APIExportEndpointSlice creation to succeed")
+
+	require.Eventually(t, func() bool {
+		slice, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(slice, apisv1alpha1.APIExportValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid APIExport")
+
+	t.Logf("Adding a Partition to the APIExportEndpointSlice")
+	slice.Spec.Partition = partition.Name
+	_, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Update(ctx, slice, metav1.UpdateOptions{})
+	require.NoError(t, err, "error updating APIExportEndpointSlice")
+
+	require.Eventually(t, func() bool {
+		slice, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsFalse(slice, apisv1alpha1.PartitionValid) && conditions.GetReason(slice, apisv1alpha1.PartitionValid) == apisv1alpha1.PartitionInvalidReferenceReason
```

Comment:
should we also examine any other conditions (if any) ?

Reply:
added check for URLs ready condition